lib/components/h264depay/index.ts

```diff
@@ -35,23 +35,33 @@ export class H264Depay extends Tube {
           msg.type === MessageType.RTP &&
           payloadType(msg.data) === h264PayloadType
         ) {
-          const endOfFrame = marker(msg.data)
           const h264Message = h264DepayParser.parse(msg)
-
-          // Skip if not a full H264 frame, or when there hasn't been an I-frame yet
-          if (
-            h264Message === null ||
-            (!idrFound && h264Message.nalType !== NAL_TYPES.IDR_PICTURE)
-          ) {
+          // Skip if not a full H264 frame
+          // Also, don't forward non-IDR frames until an IDR frame has been received
+          if (h264Message === null ||
+            (!idrFound && h264Message.nalType === NAL_TYPES.NON_IDR_PICTURE)) {
             callback()
             return
           }
 
-          idrFound = true
+          if (h264Message.nalType === NAL_TYPES.IDR_PICTURE) {
+            idrFound = true
+          }
+
+          //if (h264Message.nalType === NAL_TYPES.SPS || h264Message.nalType === NAL_TYPES.PPS) {
```

Comment:
is this commented out code required, I think there is some debug functionality I saw that could display this?

Reply:
There is a node debug module that reads settings from DEBUG env variable. Anyway, I don't view this at that important for now...